### PR TITLE
Remove ShortCircuitNetwork

### PR DESCRIPTION
**Pull request by geofjamg**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: simulator/util/src/main/java/com/powsybl/incubator/simulator/util/extensions/ShortCircuitNetworkT3W.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * Copyright (c) 2022, Jean-Baptiste Heyberger & Geoffroy Jamgotchian
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.incubator.simulator.util.extensions;
+
+import java.util.Objects;
+
+/**
```
Comment (JB-H) [inline simulator/util/src/main/java/com/powsybl/incubator/simulator/util/extensions/ShortCircuitNetworkT3W.java:11]: ShortCircuitNetworkT3W could be renamed ShortCircuitT3W in a next review

File: simulator/util/src/main/java/com/powsybl/incubator/simulator/util/extensions/ShortCircuitNetworkTransformerLeg.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.incubator.simulator.util.extensions;
+
+import java.util.Objects;
+
+/**
+ * @author Jean-Baptiste Heyberger <jbheyberger at gmail.com>
+ */
```
Comment (JB-H) [inline simulator/util/src/main/java/com/powsybl/incubator/simulator/util/extensions/ShortCircuitNetworkTransformerLeg.java:13]: Same as all other extensions, Network could be removed in a next review

**Review by JB-H — APPROVED** [review]
In line with the suppression of shortCircuitNetwork.
Removal of "Network" in extensions names could be made in a next review

**Comment by geofjamg** [comment]
> In line with the suppression of shortCircuitNetwork.
> Removal of "Network" in extensions names could be made in a next review

Done

**Pull request merged**